### PR TITLE
Dogusata/fix stream animation flows and card orders

### PR DESCRIPTION
**Pull request by dogusata**
## Problem
- `<span class='typewriter-part>...</span>` shows up in contents within list items.
- Chat card component reordering and update structure is broken sometimes depending on update type.

## Solution
- Streaming content component wrapped out from the chat-item-card and builded as a separate component
- Typewriter animation injection removed from the markdown parsing process
- Typewriter animation injection added to node process state with a check if the node type is TEXT, then we're adding the animation wrapper.
- Converted some components to exportable ones.
- Moved related `scss` file loads to their own or related components
- Updated some component's code structures to match them repo wide

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/styles/components/chat/_chat-prompt-attachment.scss [hunk 1]
```diff
@@ -1,3 +1,4 @@
+@import '../../mixins';
```
Comment (ege0zcan) [inline src/styles/components/chat/_chat-prompt-attachment.scss:1]: nit: is this import used in this css file

File: src/styles/components/chat/_chat-item-tree-view.scss [hunk 2]
```diff
@@ -196,7 +196,7 @@
         flex-flow: row-reverse nowrap;
         justify-content: flex-end;
         align-items: center;
-        gap: var(--mynah-sizing-2);
+        // gap: var(--mynah-sizing-2);
```
Comment (ege0zcan) [inline src/styles/components/chat/_chat-item-tree-view.scss:199]: is this intentional or a leftover?

File: src/helper/events.ts [hunk 3]
```diff
@@ -7,9 +7,9 @@ import { MynahEventNames } from '../static';
 import { generateUID } from './guid';
 
 export const cancelEvent = (event: Event): boolean => {
-  event.preventDefault();
-  event.stopPropagation();
-  event.stopImmediatePropagation();
+  event.preventDefault?.();
```
Comment (ege0zcan) [inline src/helper/events.ts:10]: Doesn't all events have these functions?

File: src/components/card/card-body.ts [hunk 4]
```diff
@@ -97,80 +87,93 @@ export class CardBody {
...
-            click: (e: MouseEvent) => {
-              if (this.props.onLinkClick !== undefined) {
-                this.props.onLinkClick(url, e);
-              }
-            },
-            auxclick: (e: MouseEvent) => {
-              if (this.props.onLinkClick !== undefined) {
-                this.props.onLinkClick(url, e);
-              }
+    let elementFromNode: HTMLElement = node;
```
Comment (ege0zcan) [inline src/components/card/card-body.ts:90]: nit:  this function is too big, let's consider putting logic into functions like:
```
processNode(){
 if(text.node)
   animateTextNode()
  else if(linkNode)
    linkNode()

}
```

File: src/components/syntax-highlighter.ts [hunk 5]
```diff
@@ -259,8 +244,27 @@ export class SyntaxHighlighter {
...
+              ? [ {
+                  type: 'span',
+                  classNames: [ 'mynah-syntax-highlighter-language' ],
+                  children: [ props.language ]
+                } ]
+              : []),
+            ...this.codeBlockButtons
+          ],
+        }));
+      }, 1);
```
Comment (ege0zcan) [inline src/components/syntax-highlighter.ts:266]: nit: we should explain why we have this 1ms timeout

File: src/helper/events.ts (hunk 3 above)
Comment (dogusata) [reply src/helper/events.ts:10]: Unfortunately not. It can be overridden during generation

File: src/components/card/card-body.ts (hunk 4 above)
Comment (dogusata) [reply src/components/card/card-body.ts:90]: Next plan is to move it to a helper function

**Review by ege0zcan — APPROVED** [review]
Went through changes together in a call, made small updates together. Looks good to me

**Pull request merged**